### PR TITLE
Change expr/stmt in examples to untyped/typed

### PR DESCRIPTION
**Pull request by lcrees**

### CODE REVIEW

File: examples/cross_todo/nim_commandline/nimtodo.nim [hunk 1]
```diff
@@ -65,7 +65,7 @@ proc abort(message: string, value: int) =
   quit(message, value)
 
 
-template parseTodoIdAndSetCommand(newCommand: TCommand): stmt =
+template parseTodoIdAndSetCommand(newCommand: TCommand): typed =
```
Comment (dom96) [inline examples/cross_todo/nim_commandline/nimtodo.nim]: I don't think this is correct. stmt -> untyped and expr -> typed.

**Comment by lcrees** [comment]
Changed

File: examples/cross_todo/nim_commandline/nimtodo.nim [hunk 2]
```diff
@@ -42,30 +42,27 @@ type
...
 
-
 proc abort(message: string, value: int) =
   # Simple wrapper to abort also displaying the help to the user.
   stdout.write(USAGE)
   quit(message, value)
 
-
-template parseTodoIdAndSetCommand(newCommand: TCommand): stmt =
+template parseTodoIdAndSetCommand(newCommand: Command): typed =
```
Comment (dom96) [inline examples/cross_todo/nim_commandline/nimtodo.nim]: Same mistake?

Comment (lcrees) [reply examples/cross_todo/nim_commandline/nimtodo.nim]: Is there a compelling reason not to make a 1-for-1 swap of _typed_ for _stmt_ here or should it be replaced it with _untyped_?

Comment (dom96) [reply examples/cross_todo/nim_commandline/nimtodo.nim]: I'm fairly sure ``stmt`` == ``untyped``.

Comment (lcrees) [reply examples/cross_todo/nim_commandline/nimtodo.nim]: This is the lead I followed:

https://nim-lang.org/docs/manual.html#templates-typed-vs-untyped-parameters

"Note: For historical reasons *stmt* is an alias for *typed* and *expr* an alias for *untyped*, but new code should use the newer, clearer names."

So I did a 1-to-1 replacement except where *{.immediate.}* called for *untyped*

Comment (dom96) [reply examples/cross_todo/nim_commandline/nimtodo.nim]: huh, I guess I am horribly wrong.

Comment (lcrees) [reply examples/cross_todo/nim_commandline/nimtodo.nim]: I'd use an _untyped_ in my code but significant revision of the examples beyond trimming a few legacy stylistic and syntactic is beyond the scope of this particular PR. 

I'm considering submitting PRs featuring some curated Rosetta Code examples for this directory but any proposal I make for that is still in incubation.

Comment (Araq) [reply examples/cross_todo/nim_commandline/nimtodo.nim]: Currently return types should always be ``untyped`` though.

Comment (lcrees) [reply examples/cross_todo/nim_commandline/nimtodo.nim]: Changed to untyped

**Pull request merged**